src/scripts/glue_jobs/access_requirement_snapshot.py

```
@@ -15,20 +15,24 @@ class AccessRequirementSnapshots(GlueJob):
     def __init__(self, mapping_list, partition_key):
         super().__init__(mapping_list, partition_key)
 
-    def execute(self, dynamic_frame):
-        return dynamic_frame.map(f=AccessRequirementSnapshots.transform)
+    def execute(self, dynamic_frame, logger):
+        return dynamic_frame.map(lambda record: AccessRequirementSnapshots.transform(record, logger))
 
     # Process the access requirement snapshot record
     @staticmethod
-    def transform(dynamic_record):
-        # This is the partition date
-        dynamic_record[PARTITION_KEY] = Utils.ms_to_partition_date(dynamic_record[PARTITION_KEY])
+    def transform(dynamic_record, logger):
+        try:
+            # This is the partition date
+            dynamic_record[PARTITION_KEY] = Utils.ms_to_partition_date(dynamic_record[PARTITION_KEY])
+
+            # The "duc_template_file_handle_id" is not present in every type of AR
+            if DUC_TEMPLATE_FILE_HANDLE_ID in dynamic_record:
+                # The records might come in with the syn prefix, we need to remove that
+                dynamic_record[DUC_TEMPLATE_FILE_HANDLE_ID] = Utils.syn_id_string_to_int(
+                    dynamic_record[DUC_TEMPLATE_FILE_HANDLE_ID])
+        except Exception as error:
+            logger.error("Error occurred in accessrequirementsnapshots : ", error)
```

If we swallow this exceptions, we are not going to see the errors.